--- a/net/minecraft/util/EncoderCache.java
+++ b/net/minecraft/util/EncoderCache.java
@@ -35,7 +35,7 @@
             public <T> DataResult<T> encode(A innerObject, DynamicOps<T> innerDynamicOps, T innerObject2) {
                 return EncoderCache.this.cache
                     .getUnchecked(new EncoderCache.Key<>(outerCodec, innerObject, innerDynamicOps))
-                    .map(innerObjectx -> innerObjectx instanceof Tag innerTag ? innerTag.copy() : innerObjectx);
+                    .map(innerObjectx -> (T) (innerObjectx instanceof Tag innerTag ? innerTag.copy() : innerObjectx));
             }
         };
     }
