--- a/net/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
@@ -113,7 +113,8 @@
                     Typed<?> typed3 = typed.getOrCreateTyped(opticFinder3);
                     Typed<?> typed4 = typed3.getOrCreateTyped(opticFinder4);
                     Typed<?> typed5 = typed4.getOrCreateTyped(opticFinder5);
-                    Typed<?> typed6 = Util.writeAndReadTypedOrThrow(typed5, type2, dynamic2 -> dynamic2.set("id", dynamic.createString(string)));
+                    Dynamic<?> finalDynamic = dynamic;
+                    Typed<?> typed6 = Util.writeAndReadTypedOrThrow(typed5, type2, dynamic2 -> dynamic2.set("id", finalDynamic.createString(string)));
                     typed2 = typed.set(opticFinder3, typed3.set(opticFinder4, typed4.set(opticFinder5, typed6)));
                 }
 
