--- a/net/minecraft/client/gui/spectator/categories/TeleportToPlayerMenuCategory.java
+++ b/net/minecraft/client/gui/spectator/categories/TeleportToPlayerMenuCategory.java
@@ -29,7 +29,7 @@
         this.items = collection.stream()
             .filter(playerInfo -> playerInfo.getGameMode() != GameType.SPECTATOR)
             .sorted(PROFILE_ORDER)
-            .map(playerInfo -> new PlayerMenuItem(playerInfo.getProfile()))
+            .<SpectatorMenuItem>map(playerInfo -> new PlayerMenuItem(playerInfo.getProfile()))
             .toList();
     }
 
