--- a/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ b/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -296,7 +296,7 @@
             public Entry(final Holder<FlatLevelGeneratorPreset> holder) {
                 this.preset = holder.value();
                 this.name = holder.unwrapKey()
-                    .map(resourceKey -> Component.translatable(resourceKey.location().toLanguageKey("flat_world_preset")))
+                    .<Component>map(resourceKey -> Component.translatable(resourceKey.location().toLanguageKey("flat_world_preset")))
                     .orElse(PresetFlatWorldScreen.UNKNOWN_PRESET);
             }
 
